src/main/scala/cromwell/binding/types/WdlType.scala

```diff
@@ -57,11 +59,27 @@ trait WdlType {
 object WdlType {
   val parser = new WdlParser()
 
-  private lazy val wdlTypes = Seq(
-    WdlBooleanType, WdlExpressionType, WdlFileType, WdlFloatType,
-    WdlIntegerType, WdlNamespaceType, WdlObjectType, WdlStringType
+  /* This is in the order of coercion from non-wdl types */
+  val wdlTypes: Seq[WdlType] = Seq(
+    WdlStringType, WdlIntegerType, WdlFloatType, WdlMapType(WdlAnyType, WdlAnyType),
+    WdlArrayType(WdlAnyType), WdlBooleanType, WdlObjectType,
+
+    /* Nothing should coerce to these.  Anything that WdlFileType
+     * would match would be caught by WdlStringType first.  Further
+     * coercion can happen to convert WdlStringType -> WdlFileType.
+     */
+    WdlExpressionType, WdlFileType, WdlNamespaceType
   )
 
+  def areTypesHomogeneous(values: Iterable[WdlValue]): Try[WdlType] = {
```

Comment:
Right, but how does `areTypesHomogeneous` imply that that's what is happening? I'm saying that I don't think it's correctly named. names like `areX` and `isX` imply boolean returns